packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,170 @@
+
+## Links To Relevant Work Items and Reference Material
+
+- [#873 Programmatically resize column width](https://github.com/ni/nimble/issues/873)
+- [#846 Interactively resize column width](https://github.com/ni/nimble/issues/846)
+- [Table README](./README.md)
+- [Table Design Doc](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/)
+
+
+## Implementation / Design
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+- Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
+- If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    - If a shrinking column has reached its minimum pixel size, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+- Columns can be configured to not allow a user to interactively size them
+    - The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+- Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+- Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+- Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
```

Comment:
I've added a 'Mobile considerations' section. @nate-ni, how important is it for us to drive to some agreed-upon behavior for mobile, particularly within this HLD, or is this something we would either decide on in a separate HLD, or just create an issue on it to solve later?

Reply:
At the moment, our priority is still on the desktop experience so acknowledging it in a section we can easily find later is a great way to capture it (just like you did).

If there are cases, like you mention, that have straight-forward, agreed-upon solutions and they don't significantly interfere with the time/cost of feature, then we can go ahead and include them. This will be case by case.

In this specific case for column resizing via touch, we should test it out in the initial implementations/prototypes to understand the impact of what it would take to support it better. If we find it doesn't produce too much scope creep, let's get it in. Otherwise, we will defer to a later time.